packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1609,6 +1614,11 @@ abstract class BuildContext {
   /// widgets to obtain their corresponding [InheritedElement] object so that they
   /// can call [InheritedElement.dispatchDidChangeDependencies] to actually
   /// notify the widgets that _did_ register such a relationship.
+  ///
+  /// This method should not be called from [State.deactivate] or [State.dispose]
+  /// because the element tree is no longer stable at that time. To refer to
+  /// an ancestor from one of those methods, save a reference to the ancestor
+  /// in [State.initState].
```

Comment:
For this method and all the other ones after this (i.e. all the ones that aren't inheritFromWidgetOfExactType) we should just say "This method should not be called from [State.deactivate] or [State.dispose] because the element tree is no longer stable at that time. To refer to an ancestor from one of those methods, save a reference to the ancestor by calling inheritFromWidgetOfExactType in didChangeDependencies."

This is because otherwise you might refer to an ancestor that's since gone away.